doc/develop/api/design_guidelines.rst

```diff
@@ -120,3 +127,31 @@ not implemented or optional:
   requested in the call cannot be satisfied by the implementation the call shall
   return ``-ENOTSUP``. (For example, a request for a level-triggered GPIO interrupt on
   hardware that supports only edge-triggered interrupts)
+
+The table below shows the error code conventions based on device driver use cases, but they can also be applied to other kernel components.
+
+.. list-table:: Error Codes and Meanings
```

Comment:
I don't think errno codes should be standardized like this, an errno code's meaning is tied to the specific operation which produced the error code. Just search `retval -ENODEV` in tree and you get tons of different specific definitions for the errno codes. I would not include this table in the docs.